### PR TITLE
Add acceptance tests for cnb lifecycle

### PR DESCRIPTION
**Pull request by c0d1ngm0nk3y**
### Are you submitting this PR against the [develop branch](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop)?

_All PR's to CATs should be submitted to develop and will be merged to main once they've passed acceptance._

### What is this change about?

_Describe the change and why it's needed._

Added acceptance tests using the new cloud native buildpack lifecycle.

### Please provide contextual information.

_Include any links to other PRs, stories, slack discussions, etc... that will help establish context._

* RFC for cloud native buildpacks - https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0028-cnb-lifecycle.md
* Prerequisite 1 - Changes in [cloud_controller_ng](https://github.com/cloudfoundry/cloud_controller_ng/pull/3778)
* Prerequisite 2 - Changes in [diego-release](https://github.com/cloudfoundry/diego-release/pull/929)
* Prerequisite 3 - Changes in [cf-deployment](https://github.com/cloudfoundry/cf-deployment/pull/1178)

### What version of cf-deployment have you run this cf-acceptance-test change against?

* A future version - see prerequisite 3

### Please check all that apply for this PR:

- [X] introduces a new test --- Are you sure everyone should be running this test?
- [ ] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?

- [X] YES
- [ ] N/A

### If you are introducing a new acceptance test, what is your rationale for including it CATs rather than your own acceptance test suite?

_CATs should validate common operator workflows._
_CATs is not a regression test suite._
_CATs is run by every component team to validate their releases before promotion._

Same as for docker or buildpacks lifecycle. It is some basic functionality added to `cf`.

### How many more (or fewer) seconds of runtime will this change introduce to CATs?

~ 70s

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [X] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!
_It's helpful to tag a few other folks on your team or your team alias in case we need to follow up later._

@loewenstein-sap, @modulo11, @pbusko, @nicolasbender

### CODE REVIEW

File: cnb/cnb_lifecycle.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//go:build !noInternet && !noCNB
+// +build !noInternet,!noCNB
```
Comment (ctlong) [inline cnb/cnb_lifecycle.go:2]: Why add these build tags? I did a search and see that there are a couple similar build tags, but I don't see them justified anywhere. Seems like we should lean into using our configuration parameters to skip tests we don't want to run.

**Review by ctlong — CHANGES_REQUESTED** [review]
The test LGTM!

Just a few documentation requests before I think this is ready for review.

File: cnb/cnb_lifecycle.go (hunk 1 above)
Comment (modulo11) [reply cnb/cnb_lifecycle.go:2]: We started with a copy of the docker lifecycle tests. If those tags are not used/mandatory, we can just remove them.

Comment (ctlong) [reply cnb/cnb_lifecycle.go]: Ok, AFAIK they're unused. If you could remove them please, that would be great.

Comment (ctlong) [reply cnb/cnb_lifecycle.go]: Oh seems like you did, thanks.

**Pull request merged**